DashAI/back/dependencies/registry/component_registry.py

```diff
@@ -207,6 +211,35 @@ def register_component(self, new_component: Type) -> None:
                     compatible_component,
                 )
 
+    @beartype
+    def unregister_component(self, component: Type) -> None:
+        """Remove a component from the registry.
+
+        Parameters
+        ----------
+        component : Type
+            The object to be registred.
+
+
+        """
+        base_type = self._get_base_type(component)
+
+        try:
+            self._registry[base_type].pop(component.__name__)
+            logger.info(f"Component removed: {component.__name__}")
+        except KeyError as e:
+            logger.error(
+                f"Error: Component named {component.__name__} does not exist "
+                f"in the registry. Exception: {e}"
+            )
```

Comment:
Ideally, the same exception should be raised here to avoid suppressing the exception and making it go unnoticed.

```suggestion
        except KeyError as e:
            raise ValueError(
                f"Error when unregistering a component: Component named {component.__name__} does not exist "
                f"in the registry. Exception: {e}"
            )

```

Reply:
Totally agree! But I have a question. Why we should throw ValueError instead of KeyError